src/Stratis.Features.FederatedPeg/TargetChain/CrossChainTransferStore.cs

```diff
@@ -602,6 +606,7 @@ public Task<Transaction> MergeTransactionSignaturesAsync(uint256 depositId, Tran
                                 {
                                     this.logger.LogInformation("Deposit: {0} collected enough signatures and is FullySigned", transfer.DepositTransactionId);
                                     transfer.SetStatus(CrossChainTransferStatus.FullySigned);
+                                    this.signals.Publish(new CrossChainTransferTransactionFullySigned(transfer));
```

Comment:
As discussed a transfer can also become FullySigned when the block containing the transaction is rewound. This means that transfers **don't always become fully signed in chronological order**. Are you confident that this fact is still being addressed correctly with this change?

Reply:
I'd like to know @codingupastorm opinion on this, because the issue description acceptance was just saying "Transactions are broadcasted as soon as they are FullySigned."

I think that in case of rewind, other federation member peers should already have this fully transaction, so I don't think we should re-broadcast it.

Anyway we should test this implementation of course, I'm not savvy on CCTS stuff

Reply:
@quantumagi I don't know if you read the comment above, anyway I want to add that the old code were calling GetTransfersByStatus without passing sort=true argument, so it wasn't broadcasting transaction in order.
I'd like to know what to do with this PR.
broadcasting a transaction even when the chain rewind would be as trivial as to call signals.Publish where we set the status to FullySinged, I'm just asking if we need that.
And I'd like to know more about the sorting issue, if any.

Reply:
If I understand the concern correctly:

- We might rewind, and a transaction that was SeenInBlock might go to FullySigned.
- This transaction never gets broadcasted now.

I will look into this.

Reply:
I'm looking here:

https://github.com/stratisproject/StratisBitcoinFullNode/blob/09a2d857cc44c0f91285c7377e45b518b43d080f/src/Stratis.Features.FederatedPeg/TargetChain/CrossChainTransferStore.cs#L1166

I think this code is actually incorrect - if we reorg it's likely that the transactions were created using inputs in a different order. I don't think we want to set them to FullySigned but actually to Suspended.

Reply:
I didn't implemented any change on that, but what you said seems correct to me, because suspended transaction will be validated again when the new chain will be attached, right?

I can fix it in this PR if it's ok

Reply:
I checked it out and it makes sense that it sets it to FullySigned in the end. Any that would be in a different order are dismantled in the reorg.

As such, the node could come out of a reorg with some FullySigned transactions that have not yet been broadcasted, so I think we'll need to add a call to broadcast them elsewhere as well, after RewindIfRequired I think? 

Just stewing on this one for now, will come back to it when I can

Reply:
Actually I've just been testing this and my initial thoughts were correct.

The FWM removes the transaction completely during a reorg. So even though CCTS sets it to FullySigned at that point, it gets Suspended afterwards because the transaction doesn't exist in the wallet. I'm going to push a PR with a test and change, and then this can go in @MithrilMan